util.ts

```diff
@@ -39,6 +37,42 @@ export type ResolverInfo = {
   }[];
 };
 
+export async function getBearerToken(code) {
+  const url = new URL(`${api_url}/2/oauth2/token`);
+  const params = new URLSearchParams(url.search);
+  params.append("code", code);
+  params.append("grant_type", "authorization_code");
+  params.append("client_id", state.client_id);
+  params.append("redirect_uri", `${state.endpointUrl}/callback`);
+  params.append("code_verifier", state.code_challenge);
+  const req = {
+    method: "POST",
+    body: params.toString(),
+    headers: {
+      Authorization:
+        `Basic ${Buffer.from(`${state.client_id}:${state.client_secret}`).toString('base64')}`,
+        "Content-Type": "application/x-www-form-urlencoded",
+    },
+  };
+  const res = await fetch(`${api_url}/2/oauth2/token`, req);
+  return await res.json();
+}
+
+// Parse Query String
+export const parseQS = (search: string): Record<string, string> =>
+  (search || '')
+    .replace(/^\?/g, '')
+    .split('&')
+    .reduce((acc, query) => {
+      const [key, value] = query.split('=');
+
+      if (key) {
+        acc[key] = decodeURIComponent(value);
+      }
+
+      return acc;
+    }, {} as Record<string, string>);
```

Comment:
```suggestion
export const parseQS = (qs: string): Record<string, string> =>
  Object.fromEntries(new URLSearchParams(qs).entries())
```

Something like this should work